src/containers/piercedStorage.tpp

```diff
@@ -913,6 +913,24 @@ void PiercedStorage<value_t, id_t>::rawErase(std::size_t pos, std::size_t n)
 * \param pos_second is the position of the second element that will be swapped
 */
 template<typename value_t, typename id_t>
+template<typename T, typename std::enable_if<!std::is_same<T, bool>::value>::type *>
+void PiercedStorage<value_t, id_t>::rawSwap(std::size_t pos_first, std::size_t pos_second)
+{
+    std::size_t firstOffset  = pos_first * m_nFields;
+    std::size_t secondOffset = pos_second * m_nFields;
+    for (std::size_t k = 0; k < m_nFields; ++k) {
+        std::swap(m_fields[firstOffset + k], m_fields[secondOffset + k]);
+    }
+}
+
+/**
+* Swaps two elements.
```

Comment:
"Swaps two Boolean elements", maybe?

Reply:
I've explicitly stated that the template is used when the container stores boolean values.